src/aegis_ai_web/src/feedback_logger.py

```diff
@@ -0,0 +1,128 @@
+"""
+Feedback logger module for Aegis Web API.
+
+Provides AegisLogger singleton class for thread-safe CSV feedback logging.
+"""
+
+import csv
+import fcntl
+import logging
+import os
+from datetime import datetime
+from pathlib import Path
+from typing import List, Dict
+
+from aegis_ai import __version__, get_settings
+from aegis_ai_web.src.data_models import FEEDBACK_SCHEMA
+
+logger = logging.getLogger(__name__)
+
+
+class AegisLogger:
+    """
+    Singleton class for managing feedback log file operations.
+
+    Encapsulates CSV reading and writing with thread-safe file locking.
+    """
+
+    _instance = None
+
+    def __new__(cls):
+        """Singleton pattern: return the same instance on every instantiation."""
+        if cls._instance is None:
```

Comment:
I guess this is not thread-safe.  Thinking more about it, why do we need `_instance` at all when the `AegisLogger` class does not access any data through `self`?

I have proposed a simplification in commit 52e18d67d5716225a9ca850c8f135c6219aba4b4.